### DISCUSSION TITLE
How to access Variable annotations?

### DISCUSSION THREAD

**Discussion by aniketbhoite** [square/kotlinpoet#1070] [Q&A] [answered]
This question is not specific to Kotlin poet but about javax & Kotlin Metadata
I am creating string of data class for our api optimisation like this
data class ex:
> `data class Cast(
    @Json(name = "cast_id") val castId: Int,
    val name: String,
    @Json(name = "profile_path") val profilePath: String?
)`

wanted result:
> "{\"castId\":1,\"name\":1,\"character\":1,\"profilePath\":1}"

but currently able to generate it 
> {\"cast_id\":1,\"name\":1,\"character\":1,\"profile_path\":1}

I want access @Json (or @SerializedName) annotation and its value, how do I get it? 

then Using Kotlin poet to class with this string init.

**Comment by ZacSweers** [comment, marked as answer]
This would be a better question for stackoverflow. As you mentioned, this is not specific to KotlinPoet.
